### ISSUE TITLE
Add custom `dataclass` decorator

### ISSUE THREAD

**Issue by mivade** [mivade/argparse_dataclass#8]
This is like the standard `dataclass` decorator but will also optionally (on by default) add a `parse_args` class method to the generated class. This would allow us to do shortcuts like:

```python
@dataclass
class Parameters:
    x: int
    y: int

params = Parameters.parse_args()
```

**Comment by adsharma** [comment]
https://github.com/mivade/argparse_dataclass/pull/9

Thank you. This is a really nice module.

```
from argparse_dataclass import dataclass
```

I wish this line could be more ergonomic. `args` is taken on pypi, but I see the project has been archived.

**Comment by mivade** [comment]
Closed via #9.
